src/engine.rs

```diff
@@ -822,6 +822,13 @@ impl<'a, P: ProcessRun> PubPoint<'a, P> {
                 return Ok(None)
             }
         };
+
+        if content.this_update() < Time::now() {
+            self.metrics.premature_manifests += 1;
+            warn!("{}: premature manifest", self.cert.rpki_manifest());
+            return Ok(None)
```

Comment:
Looks consistent with other checks.

That said, I lack some context I suppose but it's not clear to me why this function returns `Result<Option<ValidPointManifest>, Failed>`. As far as I can tell no Err(Failed) is ever returned. So, this feels more like an infallible function to determine whether a given manifest is eligible for use.

Reply:
Actually, it calls `validate_collected_crl` which can error.